File: frontend/src/objects/rider/RiderDeliveryStatus.ts
import type { OrderId } from '@/objects/shared/ids'

export interface RiderDeliveryStatus {
  orderId: OrderId
  assignedAt: string
  completedAt?: string
  deadlineAt: string
  wasTimeout: boolean
  timeoutExempted: boolean
  timeoutCardUsed: boolean
  overtimeSeconds: number
  canUseTimeoutCard: boolean
}

File: backend/src/rider/objects/RiderDeliveryStatus.scala
package delivery.rider.objects

import delivery.domain.OrderId

final case class RiderDeliveryStatus(
    orderId: OrderId,
    assignedAt: String,
    completedAt: Option[String],
    deadlineAt: String,
    wasTimeout: Boolean,
    timeoutExempted: Boolean,
    timeoutCardUsed: Boolean,
    overtimeSeconds: Int,
    canUseTimeoutCard: Boolean
)
